CMakeLists.txt

```diff
@@ -472,6 +470,31 @@ if (BUILD_TESTING)
     file (GLOB TEST_LD_PRELOAD "tests/LD_PRELOAD/*.c")
     add_library(allocator_overrides SHARED ${TEST_LD_PRELOAD})
 
+    # CTest configuration variables need to be set before include(CTest) is called
+    set(VALGRIND_DEFAULT " \
+        --leak-check=full \
+        --leak-resolution=high \
+        --trace-children=yes \
+        --run-libc-freeres=no \
+        -q --error-exitcode=123 \
+        --error-limit=no \
+        --num-callers=40 \
+        --undef-value-errors=no \
+        --suppressions=valgrind.suppressions")
+
+    set(VALGRIND_PENDANTIC "${VALGRIND_DEFAULT} --errors-for-leak-kinds=all")
+
+    # "pedantic valgrind" will error on memory that is "Still Reachable". 
+    # We only run this on OpenSSL 1.1.1 because there are hundreds of false positives in other libcryptos.
+    # Tracking issue: https://github.com/aws/s2n-tls/issues/4777
+    if ($ENV{S2N_LIBCRYPTO} MATCHES "openssl-1.1.1")
+        set(MEMORYCHECK_COMMAND_OPTIONS ${VALGRIND_PENDANTIC})
+    else()
+        set(MEMORYCHECK_COMMAND_OPTIONS ${VALGRIND_DEFAULT})
+    endif()
+
+    set(CTEST_MEMORYCHECK_TYPE "Valgrind")
```

Comment:
I don't think so. It should only explicitly specify we want to run Valgrind for CTest memcheck. Because CTest memcheck has some other options like "Valgrind, Purify, BoundsChecker, DrMemory, CudaSanitizer, ThreadSanitizer, AddressSanitizer, LeakSanitizer, MemorySanitizer and UndefinedBehaviorSanitizer."
https://cmake.org/cmake/help/latest/variable/CTEST_MEMORYCHECK_TYPE.html

Reply:
I have tested it again with CTest only. This command will not interfere with unit tests if we don't want to run memcheck.